zcl_foo.clas.abap

```diff
@@ -0,0 +1,25 @@
+class ZCL_FOO definition
+  public
+  final
+  create public .
+
+public section.
+
+  methods BAR
+    importing
+      !QUX type STRING .
```

Comment:
Our naming conventions are I_QUX for importing parameter. Did you run the code inspector?